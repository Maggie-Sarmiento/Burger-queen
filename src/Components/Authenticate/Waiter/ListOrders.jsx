/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import {
  Typography,
} from '@mui/material';
import React, { useState, useEffect } from 'react';
import Box from '@mui/material/Box';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
// import MoreVertIcon from '@mui/icons-material/MoreVert';
import styles from './ListOrders.module.css';

const ListOrders = () => {
  const [dataOrder, setDataOrder] = useState([]);

  useEffect(() => {
    fetch('http://localhost:3001/orders')
      .then((response) => response.json())
      .then((data) => setDataOrder(data));
  }, []);

  return (
    <section className={styles.listOrders}>
      <Box>
        <Typography
          variant="subtitle1"
          component="h2"
          sx={{
            alignSelf: 'center',
            fontWeight: 'bold',
            fontSize: '2em',
            marginBottom: '2em',
          }}
        >
          Lista de Ordenes
        </Typography>
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell><b>Mesa</b></TableCell>
                <TableCell><b>Cliente</b></TableCell>
                <TableCell><b>Status</b></TableCell>
                <TableCell><b>Productos</b></TableCell>
                <TableCell><b>Total</b></TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {dataOrder.map((order) => (
                <TableRow key={order.id}>
                  <TableCell>{order.table}</TableCell>
                  <TableCell>{order.client}</TableCell>
                  <TableCell>{order.status}</TableCell>
                  <TableCell>
                    {order.products.map(
                      (product) => <TableRow key={product.id}>{product.name}</TableRow>,
                    )}
                  </TableCell>
                  <TableCell>{order.total}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </Box>
    </section>
  );
};

export default ListOrders;
