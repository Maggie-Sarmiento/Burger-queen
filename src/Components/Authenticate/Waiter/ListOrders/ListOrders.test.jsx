/* eslint-disable no-undef */
import React from 'react';
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';
import ListOrders from './ListOrders';

describe('test ListOrders', () => {
  test('render ListOrder', () => {
    let root;
    act(() => {
      root = create(
        <ListOrders />,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
