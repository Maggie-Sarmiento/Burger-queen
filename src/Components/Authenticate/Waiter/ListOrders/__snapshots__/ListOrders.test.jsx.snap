// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ListOrders render ListOrder 1`] = `
<section
  className="listOrders"
>
  <div
    className="MuiBox-root css-0"
  >
    <h2
      className="MuiTypography-root MuiTypography-subtitle1 css-18gen10-MuiTypography-root"
    >
      Lista de Ordenes
    </h2>
    <div
      className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    >
      <table
        className="MuiTable-root css-rqglhn-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Mesa
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Cliente
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Status
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Productos
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Total
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Tiempo
              </b>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>
</section>
`;
